### PR TITLE
quick patch on nvfuser getnv/reshape to handle sequence with NumberProxy

### PR DESCRIPTION
**Pull request by jjsjann123**
1. recursively go through `getnv` on sequence to retrieve dynamic scalar for nvfuser.
2. update reshape parsing in nvfuser executor to allow NumberProxy in new shape.

### CODE REVIEW

**Comment by jjsjann123** [comment]
PR not functional yet, blocked by https://github.com/NVIDIA/Fuser/pull/2530

File: thunder/tests/test_jit_general.py
Comment (jjsjann123) [inline thunder/tests/test_jit_general.py:1106]: we might want to discuss later on how we would like to represent dynamic shape. (design review due)

Meanwhile, this is a quick patch coming from patching @kiya00 's issue: https://github.com/Lightning-AI/lightning-thunder/pull/451#issuecomment-2189428679

File: thunder/executors/nvfuserex_impl.py
Comment (IvanYashchuk) [inline thunder/executors/nvfuserex_impl.py:1201]: We don't do any type checking, but what would be a more accurate type hint for `shape`?

**Review by IvanYashchuk — APPROVED** [review]

File: thunder/tests/test_jit_general.py
Comment (kiya00) [reply thunder/tests/test_jit_general.py:1106]: Thanks for the fix, I'll try it once it merged

**Review by kiya00 — APPROVED** [review]

File: thunder/executors/nvfuserex_impl.py
Comment (jjsjann123) [reply thunder/executors/nvfuserex_impl.py:1201]: shape is going to be a `list[int, NumberProxy]` moving forward. We'll finalize that in a design review for dynamic shape :crossed_fingers:

**Review by t-vi — APPROVED** [review]
Thank you @jjsjann123 @kiya00 @IvanYashchuk

**Pull request merged**
